commit 95746ecce9215c8428a02f1745d03e10536a4129
Author: Volker Theile <vtheile@suse.com>
Date:   Mon Sep 24 12:18:56 2018 +0200

    mgr: Add ability to trigger a cluster/audit log message from Python
    
    Fixes: https://tracker.ceph.com/issues/36194
    
    Signed-off-by: Volker Theile <vtheile@suse.com>

diff --git a/src/mgr/ActivePyModules.cc b/src/mgr/ActivePyModules.cc
index 1c89dcc3188..7e22775bac7 100644
--- a/src/mgr/ActivePyModules.cc
+++ b/src/mgr/ActivePyModules.cc
@@ -38,10 +38,12 @@
 ActivePyModules::ActivePyModules(PyModuleConfig &module_config_,
           std::map<std::string, std::string> store_data,
           DaemonStateIndex &ds, ClusterState &cs,
-	  MonClient &mc, LogChannelRef clog_, Objecter &objecter_,
+          MonClient &mc, LogChannelRef clog_,
+          LogChannelRef audit_clog_, Objecter &objecter_,
           Client &client_, Finisher &f, DaemonServer &server)
   : module_config(module_config_), daemon_state(ds), cluster_state(cs),
-    monc(mc), clog(clog_), objecter(objecter_), client(client_), finisher(f),
+    monc(mc), clog(clog_), audit_clog(audit_clog_), objecter(objecter_),
+    client(client_), finisher(f),
     server(server), lock("ActivePyModules")
 {
   store_cache = std::move(store_data);
@@ -375,7 +377,7 @@ int ActivePyModules::start_one(PyModuleRef py_module)
 
   ceph_assert(modules.count(py_module->get_name()) == 0);
 
-  modules[py_module->get_name()].reset(new ActivePyModule(py_module, clog));
+  modules[py_module->get_name()].reset(new ActivePyModule(py_module, clog, audit_clog));
   auto active_module = modules.at(py_module->get_name()).get();
 
   int r = active_module->load(this);
diff --git a/src/mgr/ActivePyModules.h b/src/mgr/ActivePyModules.h
index 35476c1bbc8..48302d63ac5 100644
--- a/src/mgr/ActivePyModules.h
+++ b/src/mgr/ActivePyModules.h
@@ -42,7 +42,7 @@ class ActivePyModules
   DaemonStateIndex &daemon_state;
   ClusterState &cluster_state;
   MonClient &monc;
-  LogChannelRef clog;
+  LogChannelRef clog, audit_clog;
   Objecter &objecter;
   Client   &client;
   Finisher &finisher;
@@ -55,7 +55,7 @@ public:
   ActivePyModules(PyModuleConfig &module_config,
             std::map<std::string, std::string> store_data,
             DaemonStateIndex &ds, ClusterState &cs, MonClient &mc,
-            LogChannelRef clog_, Objecter &objecter_, Client &client_,
+            LogChannelRef clog_, LogChannelRef audit_clog_, Objecter &objecter_, Client &client_,
             Finisher &f, DaemonServer &server);
 
   ~ActivePyModules();
